diff --git a/configure.ac b/configure.ac
index 1d0b5aa6c..8bec2cdc1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -712,7 +712,7 @@ if test -n "$with_android_ndk"; then
     # (A 64-bit ld.gold grows too much over 10 gigabytes of virtual space when linking such a .so if
     # all objects have been built with debug information.)
     case $build_os in
-    linux-gnu*)
+    linux-gnu*|linuc-musl*)
         android_HOST_TAG=linux-x86_64
         ;;
     darwin*)
@@ -878,7 +878,7 @@ solaris*)
     fi
     ;;
 
-linux-gnu*|k*bsd*-gnu*|linux-musl*)
+linux-gnu*|linuc-musl*|k*bsd*-gnu*|linux-musl*)
     using_freetype_fontconfig=yes
     using_x11=yes
     build_skia=yes
@@ -5000,7 +5000,7 @@ kfreebsd*)
     esac
     ;;
 
-linux-gnu*|linux-musl*)
+linux-gnu*|linuc-musl*|linux-musl*)
     COM=GCC
     OS=LINUX
     RTL_OS=Linux
@@ -8717,7 +8717,7 @@ if test -n "$ENABLE_JAVA" -a -z "$JAVAINC"; then
         test -d "$JAVA_HOME/include/native_thread" && JAVAINC="$JAVAINC -I$JAVA_HOME/include/native_thread"
         ;;
 
-    linux-gnu*)
+    linux-gnu*|linuc-musl*)
         JAVAINC="-I$JAVA_HOME/include"
         JAVAINC="$JAVAINC -I$JAVA_HOME/include/linux"
         test -d "$JAVA_HOME/include/native_thread" && JAVAINC="$JAVAINC -I$JAVA_HOME/include/native_thread"
@@ -11388,19 +11388,6 @@ AC_SUBST(SYSTEM_LIBNUMBERTEXT)
 AC_SUBST(SYSTEM_LIBNUMBERTEXT_DATA)
 AC_SUBST(LIBNUMBERTEXT_CFLAGS)
 
-dnl ***************************************
-dnl testing libc version for Linux...
-dnl ***************************************
-if test "$_os" = "Linux"; then
-    AC_MSG_CHECKING([whether the libc is recent enough])
-    AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
-    #include <features.h>
-    #if defined(__GNU_LIBRARY__) && (__GLIBC__ < 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ < 1))
-    #error glibc >= 2.1 is required
-    #endif
-    ]])],, [AC_MSG_RESULT([yes])], [AC_MSG_ERROR([no, upgrade libc])])
-fi
-
 dnl =========================================
 dnl Check for uuidgen
 dnl =========================================
@@ -14585,7 +14572,7 @@ else
 
     case "$host_os" in
 
-    aix*|dragonfly*|freebsd*|linux-gnu*|*netbsd*|openbsd*)
+    aix*|dragonfly*|freebsd*|linux-gnu*|linuc-musl*|*netbsd*|openbsd*)
         if test "$ENABLE_JAVA" != ""; then
             pathmunge "$JAVA_HOME/bin" "after"
         fi
