diff --git a/configure.cmake b/configure.cmake
index fdcc4be..0beb9ee 100644
--- a/configure.cmake
+++ b/configure.cmake
@@ -564,9 +564,9 @@ CHECK_SYMBOL_EXISTS(EVFILT_TIMER "sys/types.h;sys/event.h;sys/time.h" HAVE_EVFIL
 IF(HAVE_KQUEUE AND HAVE_EVFILT_TIMER)
   SET(HAVE_KQUEUE_TIMERS 1 CACHE INTERNAL "Have kqueue timer-related filter")
 ELSEIF(HAVE_TIMER_CREATE AND HAVE_TIMER_SETTIME)
-  IF(HAVE_SIGEV_THREAD_ID OR HAVE_SIGEV_PORT)
+  # IF(HAVE_SIGEV_THREAD_ID OR HAVE_SIGEV_PORT)
     SET(HAVE_POSIX_TIMERS 1 CACHE INTERNAL "Have POSIX timer-related functions")
-  ENDIF()
+  # ENDIF()
 ENDIF()
 
 IF(NOT HAVE_POSIX_TIMERS AND NOT HAVE_KQUEUE_TIMERS AND NOT WIN32)
