diff --git a/sql/memory/aligned_atomic.h b/sql/memory/aligned_atomic.h
index d13d45b9..68003b4f 100644
--- a/sql/memory/aligned_atomic.h
+++ b/sql/memory/aligned_atomic.h
@@ -76,7 +76,7 @@ static inline size_t _cache_line_size() {
   return line_size;
 }
 
-#elif defined(__linux__)
+#elif defined(__linux__) && defined(__GLIBC__)
 static inline size_t _cache_line_size() {
   long size = sysconf(_SC_LEVEL1_DCACHE_LINESIZE);
   if (size == -1) return 64;
