--- mozc-afb03ddfe72dde4cf2409863a3bfea160f7a66d8/src/base/mutex.cc	2018-02-26 04:00:00.000000000 +0100
+++ mozc-python3/src/base/mutex.cc	2018-11-29 16:08:30.641514665 +0100
@@ -153,9 +153,9 @@
   // PTHREAD_MUTEX_RECURSIVE_NP but Mac OS X 10.5 does not
   pthread_mutexattr_t attr;
   pthread_mutexattr_init(&attr);
-#if defined(OS_MACOSX)
+#if defined(OS_MACOSX) || defined(OS_LINUX)
   pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_RECURSIVE);
-#elif defined(OS_LINUX) || defined(OS_ANDROID) || defined(OS_NACL)
+#elif defined(OS_ANDROID) || defined(OS_NACL)
   pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_RECURSIVE_NP);
 #else
 #error "This platform is not supported."
