diff --git a/library/base/file_functions.cpp b/library/base/file_functions.cpp
index f4c0935..9e1db83 100644
--- a/library/base/file_functions.cpp
+++ b/library/base/file_functions.cpp
@@ -28,6 +28,7 @@
 #include "base/string_utilities.h"
 
 #include <glib/gstdio.h>
+#include <fcntl.h>
 
 using namespace base;
 
@@ -260,4 +261,4 @@ void openStream(const std::string &fileName, std::wofstream &stream) {
 #endif
 }
 
-//--------------------------------------------------------------------------------------------------
\ No newline at end of file
+//--------------------------------------------------------------------------------------------------
