diff --git a/meson.build b/meson.build
index b0cd91d..51874f8 100644
--- a/meson.build
+++ b/meson.build
@@ -163,9 +163,10 @@ if cc.has_header('netinet/in.h') == true
 endif
 if cc.has_header('execinfo.h') == true
   config_h.set('HAVE_EXECINFO_H'       , '1')
-elif cc.has_function('backtrace_symbols_fd', dependencies: 'execinfo') == false
-  execinfo_dep = dependency('execinfo', required: false)
 endif
+#elif cc.has_function('backtrace_symbols_fd', dependencies: 'execinfo') == false
+#  execinfo_dep = dependency('execinfo', required: false)
+#endif
 
 if cc.has_header('fnmatch.h') == false
   error('fnmatch.h not found')
