diff --git a/meson.build b/meson.build
index 60d1899..0de9a87 100644
--- a/meson.build
+++ b/meson.build
@@ -167,8 +167,6 @@ if cc.has_header('netinet/in.h') == true
 endif
 if cc.has_header('execinfo.h') == true
   config_h.set('HAVE_EXECINFO_H'       , '1')
-elif cc.has_function('backtrace_symbols_fd', dependencies: 'execinfo') == false
-  execinfo_dep = dependency('execinfo', required: false)
 endif
 
 if cc.has_header('fnmatch.h') == false
