diff --git a/src/librustc_codegen_ssa/back/link.rs b/src/librustc_codegen_ssa/back/link.rs
index e773a7b6c..bdbaee99b 100644
--- a/src/librustc_codegen_ssa/back/link.rs
+++ b/src/librustc_codegen_ssa/back/link.rs
@@ -1595,7 +1595,7 @@ pub fn add_upstream_native_libraries(cmd: &mut dyn Linker,
             match lib.kind {
                 NativeLibraryKind::NativeUnknown => {
                     // When creating executables, match library linkage to that of the executable
-                    if crate_type == config::CrateTypeExecutable && sess.crt_static() {
+                    if crate_type == config::CrateType::Executable && sess.crt_static() {
                         cmd.link_staticlib(&name.as_str())
                     } else {
                         cmd.link_dylib(&name.as_str())
