diff --git a/src/bootstrap/builder.rs b/src/bootstrap/builder.rs
index ec7eebd..46a5ce2 100644
--- a/src/bootstrap/builder.rs
+++ b/src/bootstrap/builder.rs
@@ -940,7 +940,7 @@ impl<'a> Builder<'a> {
             let cflags = self.cflags(target).join(" ");
             cargo
                 .env(format!("CFLAGS_{}", target), cflags.clone())
-                .env("CFLAGS", cflags.clone());
+                .env("CFLAGS", cflags);
 
             if let Some(ar) = self.ar(target) {
                 let ranlib = format!("{} s", ar.display());
@@ -955,9 +955,7 @@ impl<'a> Builder<'a> {
                 let cxx = ccacheify(&cxx);
                 cargo
                     .env(format!("CXX_{}", target), &cxx)
-                    .env("CXX", &cxx)
-                    .env(format!("CXXFLAGS_{}", target), cflags.clone())
-                    .env("CXXFLAGS", cflags);
+                    .env("CXX", &cxx);
             }
         }
 
