diff --git a/src/llvm-emscripten/lib/Target/PowerPC/PPCSubtarget.cpp b/src/llvm-emscripten/lib/Target/PowerPC/PPCSubtarget.cpp
index ccf0f80c3..9c403248c 100644
--- a/src/llvm-emscripten/lib/Target/PowerPC/PPCSubtarget.cpp
+++ b/src/llvm-emscripten/lib/Target/PowerPC/PPCSubtarget.cpp
@@ -136,6 +136,10 @@ void PPCSubtarget::initSubtargetFeatures(StringRef CPU, StringRef FS) {
   if (isDarwin())
     HasLazyResolverStubs = true;
 
+  // Set up musl-specific properties.
+  if (TargetTriple.getEnvironment() == Triple::Musl)
+    SecurePlt = true;
+
   // QPX requires a 32-byte aligned stack. Note that we need to do this if
   // we're compiling for a BG/Q system regardless of whether or not QPX
   // is enabled because external functions will assume this alignment.
diff --git a/src/llvm-emscripten/lib/Target/PowerPC/PPCTargetMachine.cpp b/src/llvm-emscripten/lib/Target/PowerPC/PPCTargetMachine.cpp
index 20a83c973..39218fa82 100644
--- a/src/llvm-emscripten/lib/Target/PowerPC/PPCTargetMachine.cpp
+++ b/src/llvm-emscripten/lib/Target/PowerPC/PPCTargetMachine.cpp
@@ -214,6 +214,10 @@ static Reloc::Model getEffectiveRelocModel(const Triple &TT,
   if (TT.getArch() == Triple::ppc64 || TT.getArch() == Triple::ppc64le)
     return Reloc::PIC_;
 
+  // musl needs SecurePlt, which depends on PIC.
+  if (TT.getEnvironment() == Triple::Musl)
+    return Reloc::PIC_;
+
   // 32-bit is static by default.
   return Reloc::Static;
 }
diff --git a/src/llvm-project/llvm/lib/Target/PowerPC/PPCSubtarget.cpp b/src/llvm-project/llvm/lib/Target/PowerPC/PPCSubtarget.cpp
index c0cbfd779..5d7a021c3 100644
--- a/src/llvm-project/llvm/lib/Target/PowerPC/PPCSubtarget.cpp
+++ b/src/llvm-project/llvm/lib/Target/PowerPC/PPCSubtarget.cpp
@@ -138,6 +138,10 @@ void PPCSubtarget::initSubtargetFeatures(StringRef CPU, StringRef FS) {
   if (isDarwin())
     HasLazyResolverStubs = true;
 
+  // Set up musl-specific properties.
+  if (TargetTriple.getEnvironment() == Triple::Musl)
+    SecurePlt = true;
+
   if (HasSPE && IsPPC64)
     report_fatal_error( "SPE is only supported for 32-bit targets.\n", false);
   if (HasSPE && (HasAltivec || HasQPX || HasVSX || HasFPU))
diff --git a/src/llvm-project/llvm/lib/Target/PowerPC/PPCTargetMachine.cpp b/src/llvm-project/llvm/lib/Target/PowerPC/PPCTargetMachine.cpp
index 580d05760..ad8239a9e 100644
--- a/src/llvm-project/llvm/lib/Target/PowerPC/PPCTargetMachine.cpp
+++ b/src/llvm-project/llvm/lib/Target/PowerPC/PPCTargetMachine.cpp
@@ -218,6 +218,10 @@ static Reloc::Model getEffectiveRelocModel(const Triple &TT,
   if (TT.getArch() == Triple::ppc64)
     return Reloc::PIC_;
 
+  // musl needs SecurePlt, which depends on PIC.
+  if (TT.getEnvironment() == Triple::Musl)
+    return Reloc::PIC_;
+
   // Rest are static by default.
   return Reloc::Static;
 }
