diff --git a/src/llvm-project/llvm/include/llvm/Analysis/TargetLibraryInfo.def b/src/llvm-project/llvm/include/llvm/Analysis/TargetLibraryInfo.def
index dd1d53135..04535bc2b 100644
--- a/src/llvm-project/llvm/include/llvm/Analysis/TargetLibraryInfo.def
+++ b/src/llvm-project/llvm/include/llvm/Analysis/TargetLibraryInfo.def
@@ -27,6 +27,15 @@
 #else
 // One of TLI_DEFINE_ENUM/STRING are defined.
 
+// avoid name conflicts with musl-libc
+#undef fopen64
+#undef fseeko64
+#undef ftello64
+#undef fstat64
+#undef lstat64
+#undef stat64
+#undef tmpfile64
+
 #if defined(TLI_DEFINE_ENUM)
 #define TLI_DEFINE_ENUM_INTERNAL(enum_variant) LibFunc_##enum_variant,
 #define TLI_DEFINE_STRING_INTERNAL(string_repr)
