diff --git a/src/bootstrap/builder.rs b/src/bootstrap/builder.rs
index 0464840..05c1b67 100644
--- a/src/bootstrap/builder.rs
+++ b/src/bootstrap/builder.rs
@@ -709,7 +709,7 @@ impl<'a> Builder<'a> {
 
             let cflags = self.cflags(target).join(" ");
             cargo.env(format!("CFLAGS_{}", target), cflags.clone())
-                 .env("CFLAGS", cflags.clone());
+                 .env("CFLAGS", cflags);
 
             if let Some(ar) = self.ar(target) {
                 let ranlib = format!("{} s", ar.display());
@@ -722,9 +722,7 @@ impl<'a> Builder<'a> {
             if let Ok(cxx) = self.cxx(target) {
                 let cxx = ccacheify(&cxx);
                 cargo.env(format!("CXX_{}", target), &cxx)
-                     .env("CXX", &cxx)
-                     .env(format!("CXXFLAGS_{}", target), cflags.clone())
-                     .env("CXXFLAGS", cflags);
+                     .env("CXX", &cxx);
             }
         }
 
