--- a/build/moz.configure/rust.configure	2017-11-18 19:47:01.751415811 +0100
+++ b/build/moz.configure/rust.configure	2017-11-18 19:48:57.923776582 +0100
@@ -150,7 +150,7 @@
             ('arm', 'Linux'): 'armv7-unknown-linux-gnueabihf',
             ('sparc64', 'Linux'): 'sparc64-unknown-linux-gnu',
             ('x86', 'Linux'): 'i686-unknown-linux-gnu',
-            ('x86_64', 'Linux'): 'x86_64-unknown-linux-gnu',
+            ('x86_64', 'Linux'): 'x86_64-unknown-linux-musl',
             # OS X
             ('x86', 'OSX'): 'i686-apple-darwin',
             ('x86_64', 'OSX'): 'x86_64-apple-darwin',
