diff --git a/old-configure b/old-configure
index 19e03de..c77a027 100644
--- a/old-configure
+++ b/old-configure
@@ -10138,7 +10138,7 @@ else
   echo "$ac_t""no" 1>&6
 fi
 
-	min_nss_version=3.34.1
+	min_nss_version=3.34
 	echo $ac_n "checking for NSS - version >= $min_nss_version""... $ac_c" 1>&6
 echo "configure:10144: checking for NSS - version >= $min_nss_version" >&5
 
diff --git a/old-configure.in b/old-configure.in
index cde7b02..2d4830f 100644
--- a/old-configure.in
+++ b/old-configure.in
@@ -1908,7 +1908,7 @@ MOZ_ARG_WITH_BOOL(system-nss,
     _USE_SYSTEM_NSS=1 )
 
 if test -n "$_USE_SYSTEM_NSS"; then
-    AM_PATH_NSS(3.34.1, [MOZ_SYSTEM_NSS=1], [AC_MSG_ERROR([you don't have NSS installed or your version is too old])])
+    AM_PATH_NSS(3.34, [MOZ_SYSTEM_NSS=1], [AC_MSG_ERROR([you don't have NSS installed or your version is too old])])
 fi
 
 if test -n "$MOZ_SYSTEM_NSS"; then
