diff --git a/config/system-headers.mozbuild b/config/system-headers.mozbuild
index b257e6f8e3..236a19b754 100644
--- a/config/system-headers.mozbuild
+++ b/config/system-headers.mozbuild
@@ -279,7 +279,6 @@ system_headers = [
     'ftglyph.h',
     'ftlcdfil.h',
     'ftoutln.h',
-    'fts.h',
     'ftsizes.h',
     'ftsynth.h',
     'ftxf86.h',
diff --git a/toolkit/mozapps/update/common/updatedefines.h b/toolkit/mozapps/update/common/updatedefines.h
index e7ac01e7c3..6af55361be 100644
--- a/toolkit/mozapps/update/common/updatedefines.h
+++ b/toolkit/mozapps/update/common/updatedefines.h
@@ -102,8 +102,6 @@ static inline int mywcsprintf(WCHAR* dest, size_t count, const WCHAR* fmt,
 
 #ifdef SOLARIS
 #include <sys/stat.h>
-#else
-#include <fts.h>
 #endif
 #include <dirent.h>
 
diff --git a/toolkit/mozapps/update/updater/updater.cpp b/toolkit/mozapps/update/updater/updater.cpp
index c300260956..ad633f64fc 100644
--- a/toolkit/mozapps/update/updater/updater.cpp
+++ b/toolkit/mozapps/update/updater/updater.cpp
@@ -52,6 +52,12 @@
 #include "updaterfileutils_osx.h"
 #endif  // XP_MACOSX
 
+#ifndef SOLARIS
+#if defined(__GLIBC__)
+#include <fts.h>
+#endif
+#endif
+
 #include "mozilla/Compiler.h"
 #include "mozilla/Types.h"
 #include "mozilla/UniquePtr.h"
