diff --git a/toolkit/mozapps/update/common/updatedefines.h b/toolkit/mozapps/update/common/updatedefines.h
index e7ac01e7c3..99d549be6e 100644
--- a/toolkit/mozapps/update/common/updatedefines.h
+++ b/toolkit/mozapps/update/common/updatedefines.h
@@ -103,7 +103,7 @@ static inline int mywcsprintf(WCHAR* dest, size_t count, const WCHAR* fmt,
 #ifdef SOLARIS
 #include <sys/stat.h>
 #else
-#include <fts.h>
+# include <fts.h>
 #endif
 #include <dirent.h>
 
