diff --git a/toolkit/mozapps/update/common/updatedefines.h b/toolkit/mozapps/update/common/updatedefines.h
index e7ac01e7c3..df8e42dd89 100644
--- a/toolkit/mozapps/update/common/updatedefines.h
+++ b/toolkit/mozapps/update/common/updatedefines.h
@@ -103,8 +103,10 @@ static inline int mywcsprintf(WCHAR* dest, size_t count, const WCHAR* fmt,
 #ifdef SOLARIS
 #include <sys/stat.h>
 #else
+#if defined(__GLIBC__)
 #include <fts.h>
 #endif
+#endif
 #include <dirent.h>
 
 #ifdef XP_MACOSX
