diff --git a/configure b/configure
index 36905ea..4e2f555 100755
--- a/configure
+++ b/configure
@@ -12549,7 +12549,7 @@ else
 						     int main(){return 0;}
 _ACEOF
 if ac_fn_c_try_compile "$LINENO"; then :
-  st_cv_m32_cc=yes
+  st_cv_m32_cc=no
 else
   st_cv_m32_cc=no
 fi
@@ -12877,7 +12877,7 @@ else
 						     int main(){return 0;}
 _ACEOF
 if ac_fn_c_try_compile "$LINENO"; then :
-  st_cv_mx32_cc=yes
+  st_cv_mx32_cc=no
 else
   st_cv_mx32_cc=no
 fi
